rse-cmdline-git.Rmd

```diff
@@ -0,0 +1,3536 @@
+# Git at the command line {#cmdline-git}
+
+## Questions {#cmdline-git-questions}
+
+```{r, child="questions/cmdline-git.md"}
+```
+
+## Objectives {#cmdline-git-objectives}
+
+```{r, child="objectives/cmdline-git.md"}
+```
+
+## Introduction {#cmdline-git-intro}
+
+A version control system stores a master copy of your code in a repository,
+which you can’t edit directly.
+Instead, you checkout a working copy of the code,
+edit that code, then commit changes back to the repository.
+In this way, the system records a complete revision history (i.e. of every commit),
+so that you can retrieve and compare previous versions at any time.
+This is useful from an individual viewpoint,
+because you don’t need to store multiple (but slightly different) copies of the same script.
+
+TODO: Figure - [PhD Comics, A Story Told in File Names](http://phdcomics.com/comics.php?f=1323) 
+
+It’s also useful from a collaboration viewpoint,
+because the system keeps a record of who made what changes when
+and prevents people from overwriting other people's work.
+
+In a previous lesson (TODO: provide link),
+we learned the basics of a version control system called Git
+via a popular graphical user interface called GitKraken.
+Such interfaces are a relatively new development - originally
+Git could only be used at the command line.
```

Comment:
Not sure this is important, as this comment will quickly become outdated (GUI have existed for Git almost since it began, if very primitive).